qemu/tests/x86_cpu_model_boot.py

```diff
@@ -0,0 +1,131 @@
+import os
+import re
+import uuid
+import logging
+
+from avocado.utils import path
+from avocado.utils import process
+from virttest import virsh
+from virttest import data_dir
+from virttest import libvirt_xml
+from virttest import env_process
+from virttest import error_context
+
+
+def get_cpu_info_from_qemu_cli(name):
+    """
+    Get cpu info from qemu cmdline
+    """
+    cmd = "ps aux | grep qemu | grep -e '\s-name\s\+guest=%s\(\,\|\s\)'" % name
+    output = process.getoutput(cmd, shell=True)
+    cpu_line = re.findall("-cpu\s+(\S*)\s+", output)[0]
+    cpu_cmd = cpu_line.split(',', 1)
+    cpu_info = {}
+    cpu_info["model"] = cpu_cmd[0]
+    cpu_info["flags"] = "" if len(cpu_cmd) < 2 else cpu_cmd[1]
+    return cpu_info
+
+def get_cpu_info_from_dumpxml(name):
+    """
+    Get cpu info from virsh dumpxml
+    """
+    cpu_xml = libvirt_xml.VMXML.new_from_dumpxml(name).cpu
+    feature_list = cpu_xml.get_feature_list()
+    cpu_model = cpu_xml.model
+    cpu_features = {}
+    for i in range(0, len(feature_list)):
+        feature_name = cpu_xml.get_feature(i).get('name')
+        feature_policy = cpu_xml.get_feature(i).get('policy')
+        if feature_policy == "require":
+            feature_policy = "on"
+        elif feature_policy == "disable":
+            feature_policy = "off"
+        cpu_features[feature_name] = feature_policy
+    cpu_info = {}
+    cpu_info["model"] = cpu_model
+    cpu_info["features"] = cpu_features
+    return cpu_info
+
+def get_cpu_info_from_virsh(params):
+    """
+    Try to get cpu model and features from virsh with 'host-model'
+    """
+    vm_arch = params["vm_arch_name"]
+    if vm_arch not in ['x86_64', 'i686']:
+        raise NotImplementedError("Arch '%s' is not supported" % vm_arch)
+    machine = params.get('machine_type')
+    name = uuid.uuid4().hex
+    try:
+        path.find_command("virsh")
+    except path.CmdNotFoundError:
+        logging.warning("Virsh executable not set or found on path")
+        return
+
+    xml = """
+            <domain type='kvm'>
+                <name>%s</name>
+                <memory>1</memory>
+                <os>
+                    <type arch='%s' machine='%s'>hvm</type>
+                </os>
+                <cpu mode='host-model'/>
+            </domain>
+          """ % (name, vm_arch, machine)
+    xml_file = os.path.join(data_dir.get_tmp_dir(), "temp_xml_for_cpu")
+    with open(xml_file, "w") as f:
+        f.write(xml)
+    try:
+        logging.info("Get cpu model and features from virsh")
+        virsh.define(xml_file)
+        virsh.start(name)
+    except Exception as err:
+        logging.error(err)
+        return
+    else:
+        qemu_cpu_info = get_cpu_info_from_qemu_cli(name)
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+        if qemu_cpu_info != libvirt_cpu_info:
```

Comment:
Seems some flags name are different when they are in qemu cli and libvirt xml, sometimes with '-'  while sometimes with '_' .  So you need to convert it before comparing them.